tests/contrib/funsor/test_vectorized_markov.py

```diff
@@ -0,0 +1,450 @@
+        else pyro.markov(range(len(data)), history=history)
+    for i in markov_loop:
+        if isinstance(i, int) and i < 1:
+            x_probs = x_init
+        elif isinstance(i, int):
+            x_probs = x_trans[i-1, x_prev]
+        else:
+            x_probs = Vindex(x_trans)[(i-1)[:, None], x_prev]
+
+        x_curr = pyro.sample(
+            "x_{}".format(i), dist.Categorical(x_probs), infer={"enumerate": "parallel"})
+        with pyro.plate("tones", data.shape[-1], dim=-1):
+            pyro.sample("y_{}".format(i), dist.Normal(Vindex(locs)[..., x_curr], 1.), obs=data[i])
+        x_prev = x_curr
+
+
+#     w[t-1]      w[t]      w[t+1]
+#        |  \    ^  | \    ^   |
+#        |   \  /   |  \  /    |
+#        v    \/    v   \/     v
+#     y[t-1]  /\  y[t]  /\   y[t+1]
+#        ^   /  \   ^  /  \    ^
+#        |  /    v  | /    v   |
+#     x[t-1]      x[t]      x[t+1]
+def model_8(data, history, vectorized):
+    w_dim, x_dim, y_dim = 2, 3, 2
+    w_init = pyro.param("w_init", lambda: torch.rand(w_dim), constraint=constraints.simplex)
+    w_trans = pyro.param("w_trans", lambda: torch.rand((x_dim, w_dim)), constraint=constraints.simplex)
+    x_init = pyro.param("x_init", lambda: torch.rand(x_dim), constraint=constraints.simplex)
+    x_trans = pyro.param("x_trans", lambda: torch.rand((w_dim, x_dim)), constraint=constraints.simplex)
+    y_probs = pyro.param("y_probs", lambda: torch.rand(w_dim, x_dim, y_dim), constraint=constraints.simplex)
+
+    w_prev = x_prev = None
+    markov_loop = \
+        pyro.vectorized_markov(name="time", size=len(data), dim=-2, history=history) if vectorized \
+        else pyro.markov(range(len(data)), history=history)
+    for i in markov_loop:
+        w_curr = pyro.sample(
+            "w_{}".format(i), dist.Categorical(
+                w_init if isinstance(i, int) and i < 1 else w_trans[x_prev]),
+            infer={"enumerate": "parallel"})
+        x_curr = pyro.sample(
+            "x_{}".format(i), dist.Categorical(
+                x_init if isinstance(i, int) and i < 1 else x_trans[w_prev]),
+            infer={"enumerate": "parallel"})
+        with pyro.plate("tones", data.shape[-1], dim=-1):
+            pyro.sample("y_{}".format(i), dist.Categorical(
+                Vindex(y_probs)[w_curr, x_curr]),
+                obs=data[i])
+        x_prev, w_prev = x_curr, w_curr
+
+
+@pytest.mark.parametrize("use_replay", [True, False])
+@pytest.mark.parametrize("model,data,var,history", [
+    (model_0, torch.rand(5), "xy", 1),
+    (model_1, torch.rand(5, 4), "xy", 1),
+    (model_2, torch.ones((5, 4), dtype=torch.long), "xy", 1),
+    (model_3, torch.ones((5, 4), dtype=torch.long), "wxy", 1),
+    (model_4, torch.ones((5, 4), dtype=torch.long), "wxy", 1),
+    (model_5, torch.ones((5, 4), dtype=torch.long), "xy", 2),
+    (model_6, torch.rand(5, 4), "xy", 1),
+    (model_8, torch.ones((5, 4), dtype=torch.long), "wxy", 1),
```

Comment:
Can you add a couple new test cases to this list with the same models/variables/histories but with longer time axes (>50)? e.g.
```
(model_8, torch.ones((50, 4), dtype=torch.long), "wxy", 1)
```